File: src/main/java/models/Guide.java
package models;

import Enumeration.Role;
import Enumeration.StatutCompte;

public class Guide extends Personne {

    private String specialite;
    private String langues;
    private int experience;

    public Guide() {
        this.role = Role.GUIDE;
    }

    public Guide(int id, String nom, String prenom, String email,
                 String motDePasse, String specialite,
                 String langues, int experience) {

        super(id, nom, prenom, email, motDePasse, null,
                Role.GUIDE, StatutCompte.ACTIF, null);

        this.specialite = specialite;
        this.langues = langues;
        this.experience = experience;
    }

    public String getSpecialite() {
        return specialite;
    }

    public void setSpecialite(String specialite) {
        this.specialite = specialite;
    }

    public String getLangues() {
        return langues;
    }

    public void setLangues(String langues) {
        this.langues = langues;
    }

    public int getExperience() {
        return experience;
    }

    public void setExperience(int experience) {
        this.experience = experience;
    }


}

File: src/Entity/Guide.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\Collection;
use Doctrine\Common\Collections\ArrayCollection;
use App\Entity\Personne;
use App\Entity\Activite;

#[ORM\Entity]
class Guide
{
    #[ORM\Id]
    #[ORM\ManyToOne(targetEntity: Personne::class, inversedBy: "guides")]
    #[ORM\JoinColumn(name: 'id', referencedColumnName: 'id', onDelete: 'CASCADE')]
    private Personne $personne;

    #[ORM\Column(type: "string", length: 150)]
    private string $specialite;

    #[ORM\Column(type: "string", length: 250)]
    private string $langues;

    #[ORM\Column(type: "string")]
    private string $experience;

    #[ORM\OneToMany(mappedBy: "guide_id", targetEntity: Activite::class)]
    private Collection $activites;

    public function __construct()
    {
        $this->activites = new ArrayCollection();
    }

    // ------------------- Getters et setters -------------------
    public function getPersonne(): ?Personne
    {
        return $this->personne;
    }

    public function setPersonne(?Personne $personne): self
    {
        $this->personne = $personne;
        return $this;
    }

    public function getSpecialite(): string
    {
        return $this->specialite;
    }

    public function setSpecialite(string $specialite): self
    {
        $this->specialite = $specialite;
        return $this;
    }

    public function getLangues(): string
    {
        return $this->langues;
    }

    public function setLangues(string $langues): self
    {
        $this->langues = $langues;
        return $this;
    }

    public function getExperience(): string
    {
        return $this->experience;
    }

    public function setExperience(string $experience): self
    {
        $this->experience = $experience;
        return $this;
    }

    // ------------------- Redirection vers Personne -------------------
    public function getNom(): ?string
    {
        return $this->personne ? $this->personne->getNom() : null;
    }

    public function getPrenom(): ?string
    {
        return $this->personne ? $this->personne->getPrenom() : null;
    }

    // ------------------- Activites -------------------
    public function getActivites(): Collection
    {
        return $this->activites;
    }

    public function addActivite(Activite $activite): self
    {
        if (!$this->activites->contains($activite)) {
            $this->activites[] = $activite;
            $activite->setGuide($this);
        }
        return $this;
    }

    public function removeActivite(Activite $activite): self
    {
        if ($this->activites->removeElement($activite)) {
            if ($activite->getGuide() === $this) {
                $activite->setGuide(null);
            }
        }
        return $this;
    }
}